src/ipath/mod.rs

```diff
@@ -3,3 +3,165 @@
 //! will live in this module.
 pub mod service;
 pub mod sysfs;
+pub mod proto;
```

Comment:
"proto" is kinda overloaded (as in protobuf), but if its meaningful in your domain its whatever.